### PR TITLE
[improve][txn] Add command to abort transaction

### PR DESCRIPTION
**Pull request by hrzzzz**
Fixes #xyz

Main Issue: #xyz

PIP: #xyz 

### Motivation

In the implementation of Pulsar  transaction, if we have a stuck transaction, then the transactions after this one cannot be consumed by the consumer even if they have been committed. The consumer will be stuck until the stuck transaction is aborted due to timeout, and then it will continue to consume messages. Therefore, we need to provide an admin command to proactively abort the transaction in such situations.
### Modifications

Introduce a new API for aborting transaction.
### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

- Added test in `org.apache.pulsar.broker.admin.v3.AdminApiTransactionTest#testAbortTransaction`
- Extended integration test `org.apache.pulsar.admin.cli.PulsarAdminToolTest#transactions`

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [x] The REST endpoints
- [x] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/hrzzzz/pulsar/pull/3

### CODE REVIEW

**Comment by hrzzzz** [comment]
> @hrzzzz I think you are better to write a proposal or send a email to discuss when you are trying to introduce a new API.

OK，I will do this later

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java [hunk 1]
```diff
@@ -439,4 +439,33 @@ public void getPositionStatsInPendingAck(@Suspended final AsyncResponse asyncRes
...
+                    + "with transactionCoordinatorEnabled=true."),
+            @ApiResponse(code = 307, message = "Topic is not owned by this broker!"),
+            @ApiResponse(code = 400, message = "Topic is not a persistent topic!"),
+            @ApiResponse(code = 409, message = "Concurrent modification"),
+            @ApiResponse(code = 401, message = "This operation requires super-user access")})
+    public void abortTransaction(@Suspended final AsyncResponse asyncResponse,
+                                 @QueryParam("authoritative")
+                                 @DefaultValue("false") boolean authoritative,
+                                 @PathParam("mostSigBits") String mostSigBits,
+                                 @PathParam("leastSigBits") String leastSigBits) {
```
[the remark this answers is not in the document]
Comment (hrzzzz) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java:458]: I saw that the previous code was written like this, so I wrote it this way to ensure consistency in style

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionTest.java [hunk 2]
```diff
@@ -894,6 +897,25 @@ public void testGetPositionStatsInPendingAckStatsFroBatch() throws Exception {
...
+        Transaction transaction = pulsarClient.newTransaction()
+                .withTransactionTimeout(5, TimeUnit.MINUTES).build().get();
+
+        TxnMeta txnMeta = pulsar.getTransactionMetadataStoreService().getTxnMeta(transaction.getTxnID()).get();
+        assertEquals(txnMeta.status(), TxnStatus.OPEN);
+
+        // abort
+        admin.transactions().abortTransaction(transaction.getTxnID());
+        try {
+            pulsar.getTransactionMetadataStoreService().getTxnMeta(transaction.getTxnID()).get();
```
Comment (liangyepianzhou) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionTest.java:913]:
```suggestion
            pulsar.getTransactionMetadataStoreService().getTxnMeta(transaction.getTxnID()).get();
            fail();
```

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java [hunk 3]
```diff
@@ -400,4 +400,18 @@ PositionInPendingAckStats getPositionStatsInPendingAck(String topic, String subN
...
+
+    /**
+     * Abort a transaction.
+     *
+     * @param txnID the txnId
+     */
+    void abortTransaction(TxnID txnID) throws PulsarAdminException;
+
+    /**
+     * Asynchronously Abort a transaction.
```
Comment (liangyepianzhou) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java:412]:
```suggestion
     * Asynchronously abort a transaction.
```

**Review by Demogorgon314 — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionTest.java (hunk 2 above)
Comment (hrzzzz) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionTest.java:913]: fixed

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java (hunk 3 above)
Comment (hrzzzz) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java]: fixed

**Review by liangyepianzhou — APPROVED** [review]

**Pull request merged**